tests/integration.rs

```diff
@@ -994,7 +991,7 @@ mod integration_tests {
     /// in `spin.toml` inside `dir`.
     async fn do_test_build_command(dir: impl AsRef<Path>) -> Result<()> {
         let manifest_file = dir.as_ref().join("spin.toml");
-        let RawAppManifestAnyVersion::V1(manifest) = raw_manifest_from_file(&manifest_file).await?;
+        let manifest = raw_manifest_from_file(&manifest_file).await?.into_v1();
```

Comment:
Worth having an integration test that checks both the old and new syntaxes?

Reply:
`raw_manifest_from_file` will always map to V1New manifest since that is what `resolve_partials` always maps to. I think a the unit test i added (per your suggestion above) may be sufficient.